### PR TITLE
Fixing DQN and DDQN

### PR DESCRIPTION
**Pull request by SergioMartin86**
Fixing a problem with hyperparameter updating in DQN and DDQN
+ Now the better named udpateAgentPolicy is the function to update the policy the parameters in the agent. 
+ Updating the target of the critic is for now done at an algorithm by algorithm basis, until we figure out a better way to generalize this.

### CODE REVIEW

File: source/modules/solver/agent/discrete/DDQN/DDQN._cpp [hunk 1]
```diff
@@ -98,25 +98,29 @@ void DDQN::trainAgent()
...
-  auto aHyperparameters = _aLearner->getHyperparameters();
-
-  // Set Hyperparameter from target network
-  _aLearner->setHyperparameters(_aHatHyperparameters);
-  _vLearner->setHyperparameters(_vHatHyperparameters);
-
-  // Creating minibatch for the critic update
-  auto miniBatchIndexes = generateMiniBatch(_criticMiniBatchSize);
+  // Shuffling indexes to choose the mini batch from
+  std::shuffle(experienceReplayIndexes.begin(), experienceReplayIndexes.end(), *mt);
```
Comment (wadaniel) [inline source/modules/solver/agent/discrete/DDQN/DDQN._cpp:118]: why do we go back to this variant? or why not use generateMiniBatch?

File: examples/learning/reinforcement/cartpole/run-dqn.py [hunk 2]
```diff
@@ -4,16 +4,15 @@
...
-
 ####### Defining Korali Problem
 
 import korali
 k = korali.Engine()
 e = korali.Experiment()
 
 e["Problem"]["Type"] = "Reinforcement Learning / Discrete"
-e["Problem"]["Possible Actions"] = actions
+e["Problem"]["Possible Actions"] = [ [ -10.0 ], [ -9.0 ], [ -8.0 ], [ -7.0 ], [ -6.0 ], [ -5.0 ], [ -4.0 ], [ -3.0 ], [ -2.0 ], [ -1.0 ], [0.0],
```
[the remark this answers is not in the document]
Comment (SergioMartin86) [reply examples/learning/reinforcement/cartpole/run-dqn.py:14]: I brought it back to the original in the process of discarding possible culprits, but discovered that there was a slight mistake with the 'short' version: it generated numbers from -10 to +9. We could bring back short versions again later

File: source/modules/solver/agent/discrete/DDQN/DDQN._cpp (hunk 1 above)
Comment (SergioMartin86) [reply source/modules/solver/agent/discrete/DDQN/DDQN._cpp:118]: Yeah, I need to fix this.

File: examples/learning/reinforcement/cartpole/run-dqn.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (wadaniel) [reply examples/learning/reinforcement/cartpole/run-dqn.py:14]: ok

**Review by wadaniel — APPROVED** [review]

**Pull request merged**